contracts/masset/liquidator/Liquidator.sol

```diff
@@ -0,0 +1,303 @@
+        //address pToken = IPlatformIntegration(_integration).bAssetToPToken(_bAsset);
+        //uint256 pTokenbalance = IERC20(pToken).balanceOf(address(this));
+        //require(pTokenbalance > 0, "Unclaimed pTokens on this liquidation");
+        
+        liq.integration = _integration;
+        liq.sellToken = _sellToken;
+        liq.uniswapPath = _uniswapPath;
+        liq.amount = _amount;
+        liq.paused = _paused;
+
+        emit LiquidationUpdated(_bAsset);
+    }
+
+    /**
+    * @dev Delete a liquidation
+    * @param _bAsset The _bAsset for the liquidation
+    */
+    function deleteLiquidation(address _bAsset)
+        external
+        onlyGovernor
+    {
+        require(liquidations[_bAsset].bAsset != address(0), "No liquidation for this bAsset");
+    
+        // Check for unclaimed pTokens and revert
+        // TODO Add mocks to tests for this call
+        //address pToken = IPlatformIntegration(_integration).bAssetToPToken(_bAsset);
+        //uint256 pTokenbalance = IERC20(pToken).balanceOf(address(this));
+        //require(pTokenbalance > 0, "Unclaimed pTokens on this liquidation");
+
+        //Liquidation memory liquidation = liquidations[_bAsset];
+
+        //address sellToken = liquidation.uniswapPath[0];
+        //uint256 sellTokenBalance = IERC20(sellToken).balanceOf(address(this));
+        //require((sellTokenBalance == 0), "Sell tokens on this liquidation");
+
+        delete liquidations[_bAsset];
+        emit LiquidationDeleted(_bAsset);
+    }
+
+    /**
+    * @dev  Update the Uniswap contract address
+    *       Whilst it is unlikely this will be needed it is helpful for testing
+    * @param _uniswapAddress The uniswap address to assign
+    */
+    function updateUniswapAddress(address _uniswapAddress)
+        external
+        onlyGovernor
+    {
+        require(_uniswapAddress != address(0), "_uniswapAddress cannot be zero address");
+        uniswapAddress = _uniswapAddress;
+        emit UniswapUpdated(_uniswapAddress);
+    }
+
+    /**
+    * @dev Triggers a liquidation
+    * @param _bAsset The _bAsset liquidation to be triggered
+    */
+    function triggerLiquidation(address _bAsset)
+        external
+        payable
+    {
+        Liquidation storage liquidation = liquidations[_bAsset];
+        liquidation.lastTriggered = block.timestamp;
+
+        require(block.timestamp > liquidation.lastTriggered.add(1 days),
+                "Trigger liquidation only callable every 24 hours");
+
+        // Token being sold is the first in the Uniswap path
+        address sellToken = liquidation.uniswapPath[0];
+
+        // Token being bought is the last in the Uniswap path
+        address buyToken = liquidation.uniswapPath[liquidation.uniswapPath.length.sub(1)];
+
+        // Get allowance given by the integration contract for the sell token
+        uint256 allowance = IERC20(sellToken).allowance(liquidation.integration, address(this));
```

Comment:
If we give infinite approval from the CompoundIntegration, then this allowance will always be a high number. It is the balance we are supposed to be checking for - does the Integration have any new COMP to distribute